wtf-compiler/examples/record_downcasting.wtf

```diff
@@ -0,0 +1,88 @@
+package wtf:example@0.1.0
+
+// Record type definitions for demonstrating downcasting
+record point2d {
+    x: f32,
+    y: f32
+}
+
+record person {
+    name: string,
+    age: s32
+}
+
+// Example function demonstrating record downcasting
+export func main() -> s64 {
+    // Simple case: Create records with exact matching fields
+    let point_2d_data = {
+        x: 1.0,
+        y: 2.0
+    }
+    
+    // This works - exact field match
+    let point_2d: point2d = point_2d_data
+    
+    return 0
+}
+
+// Tests demonstrating record downcasting behavior
+test "basic record downcasting from anonymous record to named type" {
+    let anonymous_record = {
+        x: 3.0,
+        y: 4.0
+    }
+    
+    let typed_point: point2d = anonymous_record
+    // Test passes if compilation succeeds - no assertions needed
```

Comment:
assert here that the fields have the expected values.

Also, add this file to the github example workflows